ddtrace/tracer/option.go

```diff
@@ -334,7 +334,9 @@ func newConfig(opts ...StartOption) *config {
 	}
 	c.headerAsTags = newDynamicConfig("trace_header_tags", nil, setHeaderTags, equalSlice[string])
 	if v := os.Getenv("DD_TRACE_HEADER_TAGS"); v != "" {
-		WithHeaderTags(strings.Split(v, ","))(c)
+		c.headerAsTags.update(strings.Split(v, ","), originEnvVar)
```

Comment:
This just made me wonder whether the other settings that are available through RC should also get `origin: originEnvVar` on their dynamicConfig when enabled via env var, e.g, `DD_TRACE_ENABLED`

Reply:
@mtoffl01 You are right. I left this one as example and reminder to add more in a separate PR. Thanks!

Reply:
I agree with this. Ideally the constructor should set the origin.

If for some reason we can't use the constructor to do so, the generic `dynamicConfig` type should provide a method to update the startup and current values and their origins instead of manipulating the fields directly in `ddtrace/tracer/option.go`.